### PR TITLE
feat: [STO-4657]: Passing useGetExecutionDetailV2 hook to STO MFE

### PR DESCRIPTION
**Pull request by jsentongo**
### Summary

Removed unused components (`ExecutionCard` & `CardRailView`) from ExecutionSecurityView component & added UserLabel which we do need.

Added  `useGetExecutionDetailV2` to STOProjectOverview route to enable execution summary fetching within the STO MFE

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 4 Issues 
* Code issue 4 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11407)

File: src/modules/70-pipeline/interfaces/STOApp.ts [hunk 1]
```diff
@@ -6,16 +6,14 @@
...
-import type { ExecutionCardProps } from '@pipeline/components/ExecutionCard/ExecutionCard'
 import type { PipelineExecutionDetail } from 'services/pipeline-ng'
+import type { UserLabelProps } from '@common/exports'
 
 export interface STOAppCustomProps {
   stoApiPath?: string
   lang?: string
   customComponents: {
-    ExecutionCard: React.ComponentType<ExecutionCardProps>
-    CardRailView: React.ComponentType<CardRailViewProps>
```
Comment (chris-harness) [inline src/modules/70-pipeline/interfaces/STOApp.ts:18]: Don't remove these

File: src/modules/85-sto-steps/RouteDestinations.tsx [hunk 2]
```diff
@@ -159,7 +160,13 @@ const RouteDestinations: React.FC = () => {
           routes.toSTOProjectOverview({ ...accountPathProps, ...projectPathProps })
         ]}
       >
-        <ChildAppMounter ChildApp={RemoteSTOApp} customComponents={{ ExecutionCard, CardRailView }} />
+        <ChildAppMounter
+          ChildApp={RemoteSTOApp}
+          customComponents={{ ExecutionCard, CardRailView }}
+          customHooks={{
+            useGetExecutionDetailV2
```
Comment (chris-harness) [inline src/modules/85-sto-steps/RouteDestinations.tsx:167]: Don't pass API hooks through this way. Copy the API code into the sto-core codebase instead (or import it from the `@harness/microfrontend` package if that's been made available in that package)

**Comment by jsentongo** [comment]
retrigger titlecheck

**Comment by jsentongo** [comment]
Closing as rebase caused issues. Will Open a new PR

**Pull request closed without merging**